File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {

        int maxProfit = 0;
        int buy = Integer.MAX_VALUE;

        for(int price : prices){
            maxProfit = Math.max(maxProfit, price-buy);
            buy = Math.min(buy, price);
        }

        return maxProfit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:

        min_value = inf
        max_profit = 0

        for price in prices:
            if min_value > price:
                min_value = price
            else:
                max_profit = max(max_profit, price - min_value)

        return max_profit